renew-le.sh

```diff
@@ -3,35 +3,37 @@ set -o nounset -o errexit
 
 WORKDIR="/root/ipa-le"
 EMAIL=""
-#cd "$WORKDIR"
 
 ### cron
 # check that the cert will last at least 2 days from now to prevent too frequent renewal
 # comment out this line for the first run
 if [ "${1:-renew}" != "--first-time" ]
 then
-	certutil -d /etc/httpd/alias/ -V -u V -n Server-Cert -b "$(date '+%y%m%d%H%M%S%z' --date='2 days')" && exit 0
+	start_timestamp=`date +%s --date="$(openssl x509 -startdate -noout -in /var/lib/ipa/certs/httpd.crt | cut -d= -f2)" `
+	let diff=(`date +%s -d 20120203`-`date +%s -d 20120115`)/86400
+	if [ "$diff" -lt "2" ]; then
+		exit 0
+	fi
 fi
-
 # cert renewal is needed if we reached this line
 
 # cleanup
 rm -f "$WORKDIR"/*.pem
 rm -f "$WORKDIR"/httpd-csr.*
 
 # generate CSR
-certutil -R -d /etc/httpd/alias/ -k Server-Cert -f /etc/httpd/alias/pwdfile.txt -s "CN=$(hostname -f)" --extSAN "dns:$(hostname -f)" -o "$WORKDIR/httpd-csr.der"
+openssl req -new -sha256 -config "$WORKDIR/ipa-httpd.cnf"  -key /var/lib/ipa/private/httpd.key -out "$WORKDIR/httpd-csr.der"
 
 # httpd process prevents letsencrypt from working, stop it
 service httpd stop
 
 # get a new cert
 letsencrypt certonly --standalone --csr "$WORKDIR/httpd-csr.der" --email "$EMAIL" --agree-tos
 
-# remove old cert
-certutil -D -d /etc/httpd/alias/ -n Server-Cert
-# add the new cert
-certutil -A -d /etc/httpd/alias/ -n Server-Cert -t u,u,u -a -i "$WORKDIR/0000_cert.pem"
+# replace the cert
+cp /var/lib/ipa/certs/httpd.crt /var/lib/ipa/certs/httpd.crt.bkp
+mv -f ~/0000_cert.pem /var/lib/ipa/certs/httpd.crt
```

Comment:
does it have to be `~/0000_cert.pem`? In past it was in `$WORKDIR`

Reply:
It can be, fixed!